src/pages/VideoPlayer/Player.page.jsx

```diff
@@ -1,48 +1,106 @@
-import React from 'react';
+import React, { useContext } from 'react';
+import { Redirect, useParams } from 'react-router';
 import VideoCardFlat from '../../components/Common/VideoCard/video-card-flat.component';
-import formatter from '../../utils/formatter';
-import { getVideosList } from '../../utils/mock-data';
+import { SET_CURRENT_VIDEO } from '../../utils/action-types';
+import { MAX_RESUTS_ON_SEARCH } from '../../utils/constants';
+import helper from '../../utils/helpers';
+import { useYoutubeQuery } from '../../utils/hooks/useYoutubeQuery';
+import { Context } from '../../utils/store';
 import {
   Player,
   VideoListContained,
   VideoPlayerContainer,
   VideoPlayerWsrpper,
+  PlaceholderContainer,
+  Preloader,
 } from './Styled/styled-components';
 
 const VideoPlayer = () => {
+  const { videoURL } = useParams();
+  const [state, dispatch] = useContext(Context);
+
+  const { currentVideo } = state;
   window.scrollTo(0, 0);
-  return (
+
+  const { data, status } = useYoutubeQuery(state.searchValue);
+
+  const getList = () => {
```

Comment:
This whole method is almost the same as the one in Home Page, I think this could be reused as another component